### DISCUSSION TITLE
serverless有没有可能支持RPC通信？

### DISCUSSION THREAD

**Discussion by xmsz** [midwayjs/midway#1244] [Q&A] [answered]
背景
- 上游服务都是RPC通信
- Midway拿来做BFF层，且只能通过RPC通信
- 正常情况没问题，但是如果我想把Midway项目部署到serverless，则阿里云的函数计算是无法运行

然后现在遇到
- 不确定serverless能否支持http2，没找到文档
- 不确定是否有serverless rpc的方案

大家都是如何解决个需求『将BFF部署在serverless上如何进行RPC通信』

**Comment by czy88840616** [comment, marked as answer]
http2，不太可能，比较遥远。

**Reply by xmsz** [reply]
ok 那有没有什么办法可以解决『部署在serverless的程序与其他微服务进行RPC通信』 还是对方必须只能再提供http服务

**Reply by czy88840616** [reply]
我理解需要一个网关来做rpc到http协议的代理。

**Reply by xmsz** [reply]
> 我理解需要一个网关来做rpc到http协议的代理。

那就是需要再加一层去实现（我们用的微服务框架好像不支持泛化调用，所以必须再一层）

那我和同事再讨论一下，现在想法是
- 让其他微服务提供http接口，可能会被拒绝，毕竟不是他们的活
- 使用一层服务去代理、或者支持泛化调用，可能也会被拒绝，毕竟不是他们的活
- 不使用serverless，不行，因为很需要serverless

**Reply by czy88840616** [reply]
内部是有类似dubbo网关来做rpc的触发的，函数做不了是因为函数是被触发的，没流量就销毁了。

**Reply by xmsz** [reply]
> dubbo网关

ok 感谢回复
